### PR TITLE
Adding model-namespace, columns and action options to Generator

### PR DESCRIPTION
**Pull request by lk77**
Hello,

i've added some options to the DataTable Generator :

- **model-namespace** : 

> allow to overwrite the namespace set in config 
>  implictly enable --model option
> usage : --model-namespace="Models\Client\"

- **columns** :

>  allow to set custom columns
> usage : --columns="id,label,url,address"

- **action** : 

> allow to choose the view to use for column action
> usage : --action="client.action"

i've also introduced a way to use a custom datatables.stub.

if something is wrong, tell me and i will correct it.

thanks

### CODE REVIEW

File: src/Generators/stubs/datatables.stub [hunk 1]
```diff
@@ -26,7 +26,7 @@ class DummyClass extends DataTable
      */
     public function query()
     {
-        $query = ModelName::query();
+        $query = ModelName::select($this->getColumns);
```
Comment (yajra) [inline src/Generators/stubs/datatables.stub]: Missing function call here. I suggest you add a `query` before `select` for better IDE support.

```php
$query = Article::query()->select($this->getColumns());
```

File: src/Generators/DataTablesMakeCommand.php [hunk 2]
```diff
@@ -101,7 +177,10 @@ protected function replaceModelImport(&$stub)
      */
     protected function getStub()
     {
-        return __DIR__ . '/stubs/datatables.stub';
+        $config = $this->laravel['config'];
+        return $config->get('datatables.custom_template')
```
Comment (yajra) [inline src/Generators/DataTablesMakeCommand.php]: I think we should use the buttons config namespace and maybe set the key to `stub` for clarity?

```php
return $config->get('datatables-buttons.stub')
```

File: src/Generators/DataTablesMakeCommand.php [hunk 3]
```diff
@@ -128,6 +207,8 @@ protected function getOptions()
     {
         return [
             ['model', null, InputOption::VALUE_NONE, 'Use the provided name as the model.', null],
+            ['action', null, InputOption::VALUE_OPTIONAL, 'Force the use of singular in filename.', null],
```
Comment (yajra) [inline src/Generators/DataTablesMakeCommand.php]: Maybe a better description?

`Force the use of singular in filename.` 

to something like:

`Path to action column template.`

**Comment by yajra** [comment]
@lk77 as per testing the PR looks good with some minor changes needed. Looking forward to merging this. 👍

File: src/Generators/DataTablesMakeCommand.php (hunk 3 above)
Comment (lk77) [inline src/Generators/DataTablesMakeCommand.php]: Oops, i've totally messed this, it's a copy/paste of an other package, but i've forgotten to change the description

**Comment by lk77** [comment]
@yajra I've pushed your changes.

File: src/Generators/stubs/datatables.stub [hunk 4]
```diff
@@ -46,19 +46,16 @@ class DummyClass extends DataTable
...
+         * Get columns.
+         *
+         * @return array
+         */
+        protected function getColumns()
+        {
+            return [
+                'DummyColumns'
+            ];
+        }
```
Comment (yajra) [inline src/Generators/stubs/datatables.stub]: Excess indent. Please reduce one tab/4 spaces back.

File: src/Generators/DataTablesMakeCommand.php [hunk 5]
```diff
@@ -72,14 +79,83 @@ protected function getModel()
...
+    /**
+     * Parse array from definition
+     *
+     * @param  string  $definition
+     * @param  string  $delimiter
+     * @param  int     $indentation
+     * @return string
+     */
+    protected function parseArray($definition, $delimiter = ',', $indentation = 16)
+    {
```
Comment (yajra) [inline src/Generators/DataTablesMakeCommand.php]: I think 12 is the proper indention here to have a perfectly aligned columns. `$indention = 12`

**Comment by yajra** [comment]
@lk77 works like a charm as per testing. Need some more minor updates for proper code style for the generated code and this should be good to go. Thanks a lot!

**Comment by lk77** [comment]
@yajra it's done !

**Comment by yajra** [comment]
Released on v1.3.0 🚀 !

BTW, if you can. Please also send a PR for the official docs @ https://github.com/yajra/laravel-datatables-docs. Thanks!

**Comment by lk77** [comment]
Ok, it's done. PR #3 on documentation.
Tell me if something is wrong, english is not my first language.

Thanks

**Pull request merged**